staging/src/k8s.io/kubectl/pkg/cmd/expose/expose_test.go

```diff
@@ -591,7 +595,39 @@ func TestRunExposeService(t *testing.T) {
 					ClusterIP: corev1.ClusterIPNone,
 				},
 			},
-			expected: "service/foo exposed",
+			expected: "service/foo exposed (dry run)",
+			status:   200,
+		},
+		{
+			name: "namespace-yaml",
+			args: []string{"service", "baz"},
+			ns:   "testns",
+			calls: map[string]string{
+				"GET":  "/namespaces/testns/services/baz",
+				"POST": "/namespaces/testns/services",
+			},
+			input: &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{Name: "baz", Namespace: "testns", ResourceVersion: "12"},
+				Spec: corev1.ServiceSpec{
+					Selector: map[string]string{"app": "go"},
+				},
+			},
+			flags: map[string]string{"selector": "func=stream", "protocol": "UDP", "port": "14", "name": "foo", "labels": "svc=test", "type": "LoadBalancer", "dry-run": "client", "output": "yaml"},
+			output: &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "", Labels: map[string]string{"svc": "test"}},
+				Spec: corev1.ServiceSpec{
+					Ports: []corev1.ServicePort{
+						{
+							Protocol:   corev1.ProtocolUDP,
+							Port:       14,
+							TargetPort: intstr.FromInt(14),
+						},
+					},
+					Selector: map[string]string{"func": "stream"},
+					Type:     corev1.ServiceTypeLoadBalancer,
+				},
+			},
+			expected: "namespace: testns",
```

Comment:
why is this the expected output?

Reply:
Though exact result is as follows, this code checks only namespace because we are only interested in it.

```
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: null
  labels:
    svc: test
  name: foo
  namespace: testns
spec:
  ports:
  - port: 14
    protocol: UDP
    targetPort: 14
  selector:
    func: stream
  type: LoadBalancer
status:
  loadBalancer: {}
```

Reply:
@knight42 Is there any comment ?